test/Constraints/async.swift

```diff
@@ -7,3 +7,59 @@ func testNonConversions() async {
   let _: () -> Void = doAsynchronously // expected-error{{cannot convert value of type '() async -> ()' to specified type '() -> Void'}}
   let _: () async -> Void = doSynchronously // expected-error{{cannot convert value of type '() -> ()' to specified type '() async -> Void'}}
 }
+
+// Overloading
+func overloaded() -> String { "synchronous" }
+func overloaded() async -> Double { 3.14159 }
```

Comment:
Could you please add a test case where the only difference is `async` keyword just like described in the description? I think it would be useful to add a couple of tests where such function would get called and passed by value to an argument etc. to make sure that's handled correctly.

Reply:
Also it would be great to have a test where `async` overload is ranked lower e.g. it has default argument generic vs. concrete sync overload to make sure rule holds regardless of the score.

Reply:
Added more tests for these.

Reply:
Thank you!